app/models/iam_user_identity.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Subclasses the UserIdentity class in order to override the redis namespace and thereby partition
+# user sessions for va.gov from openid client applications.
+#
+class IAMUserIdentity < ::UserIdentity
+  redis_store REDIS_CONFIG[:iam_user_identity][:namespace]
+  redis_ttl REDIS_CONFIG[:iam_user_identity][:each_ttl]
+  redis_key :uuid
+
+  attribute :expiration_timestamp, Integer
+  attribute :iam_icn, String
+  attribute :iam_edipi, String
+  attribute :iam_sec_id, String
+
+  # Builds an identity instance from the profile returned in the IAM introspect response
+  #
+  # @param iam_profile [Hash] the profile of the user, as they are known to the IAM SSOe service.
+  # @return [IAMUserIdentity] an instance of this class
+  #
+  def self.build_from_iam_profile(iam_profile)
+    loa_level = iam_profile[:fediamassur_level].to_i
+
+    identity = new(
+      email: iam_profile[:email],
+      expiration_timestamp: iam_profile[:exp],
+      first_name: iam_profile[:given_name],
+      iam_icn: iam_profile[:fediam_mviicn],
+      iam_edipi: iam_profile[:fediam_do_dedipn_id],
+      iam_sec_id: iam_profile[:fediamsecid],
+      last_name: iam_profile[:family_name],
+      loa: { current: loa_level, highest: loa_level },
```

Comment:
I think this is ok, but want to confirm. loa_level will always be "3" from IAM SSOe?

Reply:
👍 I'll double check that with Perry.